File: S4-Systems-for-Design-and-Impl/Labs/Lab9/ProiectCSharp/TransportServer/StartRpcServer.cs
using System;
using System.Collections.Generic;
using System.Configuration;

using TransportNetworking.transport.network.utils;
using TransportPersistence.transport.persistence;
using TransportServer.transport.server;
using TransportServices.transport.services;

namespace TransportServer
{
    public class StartRpcServer
    {
        private static int defaultPort = 55555;
        private static String defaultId = "127.0.0.1";

        /*[STAThread]*/
        public static void Main(string[] args)
        {
            IReservationManagerRepository reservationManagerRepository = new ReservationManagerDBRepository();
            IReservationRepository reservationRepository = new ReservationDBRepository();
            ITripRepository tripRepository = new TripDBRepository();
            ISeatRepository seatRepository = new SeatDBRepository();
            ITransportServices transportServerImpl = new TransportServiceImpl(reservationManagerRepository, reservationRepository, tripRepository, seatRepository);

            int transportServerPort = defaultPort;
            try
            {
                transportServerPort = int.Parse(ConfigurationManager.AppSettings["TransportServer.Port"]);
            }
            catch (FormatException nef)
            {
                Console.Error.WriteLine("Wrong Port Number" + nef.Message);
                Console.Error.WriteLine("Using default port " + defaultPort);
            }
            Console.WriteLine("Starting server on port: " + transportServerPort);
            AbstractServer server = new TransportRpcConcurrentServer(transportServerPort, transportServerImpl);
            try
            {
                server.Start();
            }
            catch (ServerException e)
            {
                Console.Error.WriteLine("Error starting the server" + e.Message);
            }
            finally
            {
                try
                {
                    server.Stop();
                }
                catch (ServerException e)
                {
                    Console.Error.WriteLine("Error stopping server " + e.Message);
                }
            }
        }
    }
}

File: Systems-for-Design-and-Implementation/Labs/Lab11/ProiectJava/TransportServer/src/main/java/StartRpcServer.java
import transport.network.utils.AbstractServer;
import transport.network.utils.ServerException;
import transport.network.utils.TransportRpcConcurrentServer;
import transport.persistence.IReservationManagerRepository;
import transport.persistence.IReservationRepository;
import transport.persistence.ISeatRepository;
import transport.persistence.ITripRepository;
import transport.persistence.repository.*;
import transport.server.TransportServicesImpl;
import transport.services.ITransportServices;

import java.io.IOException;
import java.util.Properties;

public class StartRpcServer {
    private static int defaultPort = 55555;
    public static void main(String[] args) {

        Properties serverProps = new Properties();
        try {
            serverProps.load(StartRpcServer.class.getResourceAsStream("/transportserver.properties"));
            System.out.println("Server properties set. ");
            serverProps.list(System.out);
        } catch (IOException e) {
            System.err.println("Cannot find transportserver.properties "+e);
            return;
        }
        IReservationManagerRepository reservationManagerRepository = new ReservationManagerDBRepository(serverProps);
        IReservationRepository reservationRepository = new ReservationDBRepository(serverProps);
        ITripRepository tripRepository = new TripDBRepository(serverProps);
//        ITripRepository tripRepository = new TripDBRepositoryHibernate();
        ISeatRepository seatRepository = new SeatDBRepository(serverProps);
        ITransportServices transportServerImpl = new TransportServicesImpl(reservationManagerRepository, reservationRepository, tripRepository, seatRepository);
        int transportServerPort = defaultPort;
        try {
            transportServerPort = Integer.parseInt(serverProps.getProperty("transport.server.port"));
        } catch (NumberFormatException nef){
            System.err.println("Wrong  Port Number" + nef.getMessage());
            System.err.println("Using default port "+ defaultPort);
        }
        System.out.println("Starting server on port: " + transportServerPort);
        AbstractServer server = new TransportRpcConcurrentServer(transportServerPort, transportServerImpl);
        try {
            server.start();
        } catch (ServerException e) {
            System.err.println("Error starting the server" + e.getMessage());
        } finally {
            try {
                server.stop();
            } catch(ServerException e){
                System.err.println("Error stopping server "+e.getMessage());
            }
        }
    }
}